android/src/main/kotlin/com/abilia/system_settings_editor/SystemSettingsEditorPlugin.kt

```diff
@@ -8,28 +15,116 @@ import io.flutter.plugin.common.MethodChannel
 import io.flutter.plugin.common.MethodChannel.MethodCallHandler
 import io.flutter.plugin.common.MethodChannel.Result
 
-/** SystemSettingsEditorPlugin */
-class SystemSettingsEditorPlugin: FlutterPlugin, MethodCallHandler {
-  /// The MethodChannel that will the communication between Flutter and native Android
-  ///
-  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
-  /// when the Flutter Engine is detached from the Activity
+import android.provider.Settings
+import android.util.Log
+import android.view.WindowManager
+import androidx.annotation.RequiresApi
+import androidx.core.content.ContextCompat.startActivity
+import io.flutter.embedding.engine.plugins.activity.ActivityAware
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
+
+class SystemSettingsEditorPlugin: FlutterPlugin, MethodCallHandler, ActivityAware {
   private lateinit var channel : MethodChannel
+  private lateinit var context : Context
+  private lateinit var activity : Activity
 
   override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
     channel = MethodChannel(flutterPluginBinding.binaryMessenger, "system_settings_editor")
     channel.setMethodCallHandler(this)
+    context = flutterPluginBinding.applicationContext
+  }
+
+  override fun onAttachedToActivity(binding: ActivityPluginBinding) {
+    this.activity = binding.activity;
   }
 
+  @RequiresApi(Build.VERSION_CODES.M)
   override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
-    if (call.method == "getPlatformVersion") {
-      result.success("Android ${android.os.Build.VERSION.RELEASE}")
+    if (!Settings.System.canWrite(context)) {
+      openAndroidPermissionsMenu()
```

Comment:
No sure opening ACTION_MANAGE_WRITE_SETTINGS is the responsibility for
1. (hopefully) It is not gonna be necessary in MP
2. better to have an API entry for opening ACTION_MANAGE_WRITE_SETTINGS and Settings.System.canWrite(context) 

But really this is the best start, before we know the installation process and how the premissions are set